src/wallet/wallet.cpp

```diff
@@ -2660,6 +2660,12 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
 
             // Get the fee rate to use effective values in coin selection
             CFeeRate nFeeRateNeeded = GetMinimumFeeRate(*this, coin_control, &feeCalc);
+            // Do not, ever, assume that it's fine to change the fee rate if the user has explicitly
+            // provided one
+            if (coin_control.m_feerate && nFeeRateNeeded > *coin_control.m_feerate) {
+                strFailReason = strprintf("Fee rate is too low for the current mempool. Increase it to %s or modify -maxmempool", nFeeRateNeeded.ToString(coin_control.m_fee_mode));
```

Comment:
Actually I find this `-maxmempool` advice pretty dangerous for users. It's going to simply result in stuck transactions, no?

Reply:
I think it might get stuck for the user themselves, but unless their peers also increase their mempool sizes, it won't propagate very far. But I'm now a bit hesitant to suggest using it as it'll probably not do what people expect. (As you say, it will get stuck in the sense that it won't get mined or propagate properly..)

Reply:
That's something we can point out in `-maxmempool` documentation. Once #18038 lands the mempool will at least hold on to the transaction if you unload a wallet.

Reply:
Agree that we shouldn't mention -maxmempool here; it indeed likely will not do what users expect it to do.

Reply:
Changed to "Fee rate is too low for the current mempool. Increase it to %s or wait."